nomad/api/exceptions.py

```diff
@@ -1,10 +1,16 @@
+import requests
+
+
 class BaseNomadException(Exception):
     """General Error occurred when interacting with nomad API"""
     def __init__(self, nomad_resp):
         self.nomad_resp = nomad_resp
 
     def __str__(self):
-        return 'The {0} was raised with following response: {1}.'.format(self.__class__.__name__, self.nomad_resp.text)
+        if isinstance(self.nomad_resp, requests.Response) and hasattr(self.nomad_resp, "text"):
```

Comment:
It does seem rather funky to have this logic be part of the exception itself. I am tempted to suggest it would be better to tackle things from the other direction. For example something like:

* Have `BaseNomadException` just be a stock python `Exception`
* Add a `NomadRequestException` class that can include this response.text printing logic
* Update https://github.com/cloudbuy/python-nomad/blob/e10abe4191d96a0715c29eb00bbc2f7051af2d15/nomad/api/base.py#L159 to use `NomadRequestException`
* Leave https://github.com/cloudbuy/python-nomad/blob/e10abe4191d96a0715c29eb00bbc2f7051af2d15/nomad/api/base.py#L162 as-is

However I also feel like that sort of change introduces its own fiddly bits for various reasons so I can live with this route :)